sdk-extensions/jaeger-remote-sampler/src/main/java/io/opentelemetry/sdk/extension/trace/jaeger/sampler/JaegerRemoteSampler.java

```diff
@@ -63,7 +74,15 @@ public SamplingResult shouldSample(
       SpanKind spanKind,
       Attributes attributes,
       List<LinkData> parentLinks) {
-    return sampler.shouldSample(parentContext, traceId, name, spanKind, attributes, parentLinks);
+    SamplingResult samplingResult =
+        sampler.shouldSample(parentContext, traceId, name, spanKind, attributes, parentLinks);
+    // If there are no attributes in the result, then the sampler is not PerOperation based. Add
+    // the service level result attributes only if this is a root span (i.e) no valid parent context
+    if (!Span.fromContext(parentContext).getSpanContext().isValid()
```

Comment:
Sorry. I did not completely follow the comment. The if condition is to add the attributes only if both the below conditions are met
1. If there are no existing attributes. If the decision was take by say TraceIdBasedRatio sampler or any other parent sampler the attributes will not be present. 
2 And this is a the root span. If this is not a root span, the the ParentBased sampler would have been exercised (Either AlwayOn or AlwaysOff). Both will not add the attributes.

Reply:
I was referring to the check for root span - I don't think it should affect the decision to expose sampler attributes. If a sampler makes a decision, especially YES decision, it should populate the attributes to identify itself, regardless of the span being root or not. ParentBased sampler may be the only exception to this since it's not really making a decision.

Reply:
The "sampler" is a parent based sampler.